CRM/Contribute/Import/Parser/Contribution.php

```diff
@@ -257,42 +215,12 @@ public function import($onDuplicate, &$values) {
 
     $params = &$this->getActiveFieldParams();
     $formatted = ['version' => 3, 'skipRecentView' => TRUE, 'skipCleanMoney' => FALSE];
-    $dateType = CRM_Core_Session::singleton()->get('dateTypes');
-
-    $customDataType = !empty($params['contact_type']) ? $params['contact_type'] : 'Contribution';
-    $customFields = CRM_Core_BAO_CustomField::getFields($customDataType);
 
     //CRM-10994
     if (isset($params['total_amount']) && $params['total_amount'] == 0) {
       $params['total_amount'] = '0.00';
     }
-    foreach ($params as $key => $val) {
-      if ($val) {
-        switch ($key) {
-          case 'receive_date':
-          case 'cancel_date':
-          case 'receipt_date':
-          case 'thankyou_date':
-            $params[$key] = CRM_Utils_Date::formatDate($params[$key], $dateType);
-            break;
-
-          case 'pledge_payment':
-            $params[$key] = CRM_Utils_String::strtobool($val);
-            break;
-
-        }
-        if ($customFieldID = CRM_Core_BAO_CustomField::getKeyID($key)) {
-          if ($customFields[$customFieldID]['data_type'] == 'Date') {
-            CRM_Contact_Import_Parser_Contact::formatCustomDate($params, $formatted, $dateType, $key);
-            unset($params[$key]);
```

Comment:
**comment 2**: looks like when you refactored this function, you are not keeping the var `$formatted` with all the values.. so in this case  `unset($params[$key]);` don't harm because the value of the custom field is in `$formatted` but in the new function `formatInput(&$params)`  is lost.. maybe adding `$formatted`as second parameter?  
`public function formatInput(&$params, &$formatted)`